### PR TITLE
CATTY-10 add cancel button for PickerViewer

### PR DESCRIPTION
**Pull request by aeluna**
In addition to the "Done" button for the pull downs for variables, lists, and broadcast messages in the corresponding bricks, there should be a "Cancel" button which cancels this action (e.g., if opened accidentally). 

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/Catty UITests/ScriptCollectionView/BrickPickerViewerTest.swift [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+         app.collectionViews.cells.otherElements.containing(.staticText, identifier: kLocalizedBroadcast).children(matching: .other).element.tap()
+
+         app.pickerWheels.element.adjust(toPickerWheelValue: testVar3)
+         app.buttons[kLocalizedDone].tap()
+
+         XCTAssert(waitForElementToAppear(app.otherElements["MessageView_" + testVar3]).exists)
+    }
+
+    /* Uncomment, when bug with BroadCastBrick, BroadcastAndWaitBrick and
```
Comment (roelac11) [inline src/Catty UITests/ScriptCollectionView/BrickPickerViewerTest.swift]: maybe add TODO in the comment

File: src/Catty UITests/ScriptCollectionView/BrickPickerViewerTest.swift [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+         app.collectionViews.cells.otherElements.containing(.staticText, identifier: kLocalizedBroadcastAndWait).children(matching: .other).element.tap()
+
+         app.pickerWheels.element.adjust(toPickerWheelValue: testVar3)
+         app.buttons[kLocalizedDone].tap()
+
+         XCTAssert(waitForElementToAppear(app.otherElements["MessageView_" + testVar3]).exists)
+    }
+
+    /*func testChangeValueCancelForBroadcastAndWaitBrick() {
```
Comment (roelac11) [inline src/Catty UITests/ScriptCollectionView/BrickPickerViewerTest.swift]: add comment description as above

File: src/Catty UITests/ScriptCollectionView/BrickPickerViewerTest.swift [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+
+         app.collectionViews.cells.otherElements.containing(.staticText, identifier: kLocalizedWhenYouReceive).children(matching: .other).element.tap()
+
+         app.pickerWheels.element.adjust(toPickerWheelValue: testVar3)
+         app.buttons[kLocalizedDone].tap()
+
+         XCTAssert(waitForElementToAppear(app.otherElements["MessageView_" + testVar3]).exists)
+    }
+
+    /*func testChangeValueCancelForkLocalizedWhenYouReceiveBrick() {
```
Comment (roelac11) [inline src/Catty UITests/ScriptCollectionView/BrickPickerViewerTest.swift]: add comment description as above

File: src/Catty/Views/Custom/ComboBox/iOSCombobox.m [hunk 4]
```diff
@@ -266,6 +277,11 @@ - (void)setCurrentImagee:(UIImage *)image {
     [self setNeedsDisplay];
 }
 
+- (void)setChangedImagee:(UIImage *)image {
```
[the remark this answers is not in the document]
Comment (aeluna) [reply src/Catty/Views/Custom/ComboBox/iOSCombobox.m]: There is already a method named "setChangedImage" / "setCurrentImage". The method also already existed for "setCurrentImagee", so I just replicated the given structure. :)

File: src/Catty/Views/Custom/ComboBox/BSKeyboardControls.h [hunk 5]
```diff
@@ -106,6 +117,7 @@ typedef enum
  */
 @property (nonatomic, strong) UIColor *doneTintColor;
 
+
```
[the remark this answers is not in the document]
Comment (aeluna) [reply src/Catty/Views/Custom/ComboBox/BSKeyboardControls.h]: Sure. Thank you for the hint.

File: src/Catty/Views/Custom/ComboBox/BSKeyboardControls.m [hunk 6]
```diff
@@ -76,12 +82,14 @@ - (void)dealloc
...
     [self setBarTintColor:nil];
     [self setNextTitle:nil];
+    [self setCancelTitle:nil];
+    [self setCancelTintColor:nil];
     [self setDoneTitle:nil];
     [self setDoneTintColor:nil];
     [self setActiveField:nil];
     [self setToolbar:nil];
     [self setSegmentedControlItem:nil];
-    [self setDoneButton:nil];
```
[the remark this answers is not in the document]
Comment (aeluna) [reply src/Catty/Views/Custom/ComboBox/BSKeyboardControls.m]: It is necessary, so thank you for finding this! 👍

File: src/Catty UITests/ScriptCollectionView/BrickPickerViewerTest.swift (hunk 1 above)
Comment (aeluna) [reply src/Catty UITests/ScriptCollectionView/BrickPickerViewerTest.swift]: Yep, good idea! :)

**Pull request merged**